wp-admin/js/widgets/media-widgets.js

```diff
@@ -134,13 +160,13 @@ wp.mediaWidgets = ( function( $ ) {
 						});
 
 						embedLinkView.dfd.done( function( response ) {
-							embedLinkView.renderoEmbed( {
-								data: {
-									body: response.html
-								}
-							});
+							if ( embedLinkView.controller.options.mimeType !== response.type ) {
+								embedLinkView.renderFail();
```

Comment:
In any case, the video widget only supports YouTube and Vimeo. The audio widget doesn't support oEmbed at all: you can _only_ provide URLs to audio files. This would need to be revisited for adding additional media types.